Libraries/Opc.Ua.Security.Certificates/Extensions/X509AuthorityKeyIdentifierExtension.cs

```diff
@@ -0,0 +1,340 @@
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * The complete license agreement can be found here:
+ * http://opcfoundation.org/License/MIT/1.00/
+ * ======================================================================*/
+
+using System;
+using System.Formats.Asn1;
+using System.Numerics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Opc.Ua.Security.Certificates
+{
+    /// <summary>
+    /// Stores the authority key identifier extension.
+    /// </summary>
+    /// <remarks>
+    ///     id-ce-authorityKeyIdentifier OBJECT IDENTIFIER ::=  { id-ce 35 }
+    ///     AuthorityKeyIdentifier ::= SEQUENCE {
+    ///         keyIdentifier[0] KeyIdentifier           OPTIONAL,
+    ///         authorityCertIssuer[1] GeneralNames            OPTIONAL,
+    ///         authorityCertSerialNumber[2] CertificateSerialNumber OPTIONAL
+    ///         }
+    ///     KeyIdentifier::= OCTET STRING
+    /// </remarks>
+    public class X509AuthorityKeyIdentifierExtension : X509Extension
+    {
+        #region Constructors
+        /// <summary>
+        /// Creates an empty extension.
+        /// </summary>
+        protected X509AuthorityKeyIdentifierExtension()
+        {
+        }
+
+        /// <summary>
+        /// Creates an extension from ASN.1 encoded data.
+        /// </summary>
+        public X509AuthorityKeyIdentifierExtension(AsnEncodedData encodedExtension, bool critical)
+        :
+            this(encodedExtension.Oid, encodedExtension.RawData, critical)
+        {
+        }
+
+        /// <summary>
+        /// Creates an extension from ASN.1 encoded data.
+        /// </summary>
+        public X509AuthorityKeyIdentifierExtension(string oid, byte[] rawData, bool critical)
+        :
+            this(new Oid(oid, kFriendlyName), rawData, critical)
+        {
+        }
+
+        /// <summary>
+        /// Build the X509 Authority Key extension.
+        /// </summary>
+        /// <param name="subjectKeyIdentifier">The subject key identifier</param>
+        public X509AuthorityKeyIdentifierExtension(
+            byte[] subjectKeyIdentifier
+            )
+        {
+            if (subjectKeyIdentifier == null) throw new ArgumentNullException(nameof(subjectKeyIdentifier));
+            m_keyIdentifier = subjectKeyIdentifier;
+            base.Oid = new Oid(AuthorityKeyIdentifier2Oid, kFriendlyName);
+            base.Critical = false;
+            base.RawData = Encode();
+        }
+
+        /// <summary>
+        /// Build the X509 Authority Key extension.
+        /// </summary>
+        /// <param name="subjectKeyIdentifier">The subject key identifier</param>
+        /// <param name="authorityName">The distinguished name of the issuer</param>
+        /// <param name="serialNumber">The serial number of the issuer certificate</param>
```

Comment:
Based on the usage of this value it's supposed to be little endian.  Probably worth documenting.  Or changing the code so it's big-endian.

Reply:
fixed..